vxquery-core/src/test/java/org/apache/vxquery/datamodel/ObjectByteTest.java

```diff
@@ -0,0 +1,184 @@
+    ArrayBackedValueStorage abvsResult = new ArrayBackedValueStorage();
+    ObjectBuilder ob = new ObjectBuilder();
+    TaggedValuePointable tvp = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp1 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp2 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp3 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp4 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp5 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    TaggedValuePointable tvp6 = (TaggedValuePointable) TaggedValuePointable.FACTORY.createPointable();
+    SequencePointable sp = (SequencePointable) SequencePointable.FACTORY.createPointable();
+    ObjectPointable op = (ObjectPointable) ObjectPointable.FACTORY.createPointable();
+
+    @Test
+    public void testEmptyObject() {
+        // Build test Object
+        abvsResult.reset();
+        try {
+            ob.reset(abvsResult);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the Object pointable.");
+        }
+        tvp.set(abvsResult);
+
+        // Check results.
+        if (tvp.getTag() != ValueTag.OBJECT_TAG) {
+            Assert.fail("Type tag is incorrect. Expected: " + ValueTag.OBJECT_TAG + " Got: " + tvp.getTag());
+        }
+        tvp.getValue(op);
+        if (op.getEntryCount() != 0) {
+            Assert.fail("Object size is incorrect. Expected: 0 Got: " + op.getEntryCount());
+        }
+    }
+
+    @Test
+    public void testSingleItemObject() {
+        // Build test Object
+        abvsResult.reset();
+        try {
+            ob.reset(abvsResult);
+            getTaggedValuePointable("id", tvp1);
+            getTaggedValuePointable(1, tvp2);
+            ob.addItem(tvp1, tvp2);
+            ob.finish();
+        } catch (IOException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+        tvp.set(abvsResult);
+
+        // Check results.
+        if (tvp.getTag() != ValueTag.OBJECT_TAG) {
+            Assert.fail("Type tag is incorrect. Expected: " + ValueTag.OBJECT_TAG + " Got: " + tvp.getTag());
+        }
+
+        tvp.getValue(op);
+        if (op.getEntryCount() != 1) {
+            Assert.fail("Object size is incorrect. Expected: 1 Got: " + op.getEntryCount());
+        }
+        try {
+            op.getKeys(tvp);
+        } catch (SystemException e) {
+            Assert.fail("Test failed to write the object pointable.");
+        }
+
+        if (tvp.getTag() != ValueTag.XS_STRING_TAG) {
+            Assert.fail("Type tag is incorrect. Expected: " + ValueTag.XS_STRING_TAG + " Got: " + tvp.getTag());
+        }
+        if (!FunctionHelper.arraysEqual(tvp, tvp1)) {
```

Comment:
Give tvp1 names that describe the data it represents: key or value (tvpKey1, etc.).